src/fabian_nemazi_ex/pa02/test_pa02.py

```diff
@@ -0,0 +1,145 @@
+__author__ = 'Fabian Nemazi', 'Ahmed salaad Ahmed.'
+__email__ = 'Salmaaah@nmbu.no', 'fabinema@nmbu.no'
+
+import chutes_simulation as cs
+import pytest
+
+class TestBoard:
+    """
+    test for board class
+    """
+    def test_constructor(self):
+        """
+        making sure you cant give invalid ladders and chutes input
+        """
+        b = cs.Board()
+        assert(b.ladders_and_snakes == {1: 40, 8: 10, 36: 52, 43: 62, 49: 79,
+                                        65: 82, 68: 85, 24: 5, 33: 3,
+                                        42: 30, 56: 37,
+                                        64: 27, 74: 12, 87: 70})
+        c = cs.Board(ladders=[(3, 4), (5, 6)], chutes=[(10, 3), (44, 2)])
+        assert(c.ladders_and_snakes == {3: 4, 5: 6, 10: 3, 44: 2})
+
+    def test_position_adjustment(self):
+        """
+        checks that the adjustment function gives out correct input
+        """
+        b = cs.Board()
+        assert(b.position_adjustment(1) == 39)
+        assert(b.position_adjustment(0) == 0)
+
+
+class TestPlayer:
+    """
+    test for player class
+    """
+    def test_moves(self):
+        """
+        testing that the function works properly
+        """
+        b = cs.Board()
+        p = cs.Player(b)
+        for _ in range(5):
+            p.move()
+        assert(p.moves == 5)
+        assert(p.position > 0)
+
+
+class TestResilientPlayer:
+    """
+    test for resilient player class
+    """
+    def test_move(self):
+        """
+        checks that the extra step is in play, if extra steps is 6
+        the player cant have moved shorter than to step 7.
+        First chute starts at 5, therefore after first move
+        the player must minimum be at 7
+        """
+        b = cs.Board()
+        p = cs.ResilientPlayer(b, extra_steps = 6)
+        p.move()
+        first_pos = p.position
+        assert(first_pos >= 7)
```

Comment:
Requirement misunderstood leads to this wrong test. (But this tests exactly what you did 👍 )